ge/org.osate.ge.fx/src/org/osate/ge/fx/palette/PaletteGroup.java

```diff
@@ -0,0 +1,60 @@
+package org.osate.ge.fx.palette;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+
+/**
+ * Represents a single group in the palette.
+ *
+ */
+class PaletteGroup extends Region {
+	private BooleanProperty expanded = new SimpleBooleanProperty(false);
+
+	public PaletteGroup(final PaletteModelController mc, final int groupIndex) {
+		// TODO: Populate contents based in a manner similar to what is currently in Palette
+
+		VBox buttonBox = new VBox();
+
+		Button groupButton = new Button(mc.getGroupLabel(groupIndex));
+		groupButton.setGraphic(new ImageView(mc.getGroupIcon(groupIndex)));
+
+		buttonBox.getChildren().add(groupButton);
+
+		ScrollPane scrollPane = new ScrollPane();
+		for (int i = 0; i < mc.getNumberOfItems(); i++) {
+
+			if (mc.getItemGroup(i) != null && mc.getItemGroup(i) == groupIndex) {
+
+				Button itemButton = new Button(mc.getItemLabel(i));
```

Comment:
Create a PaletteItem class which is a region that represents an item. Pass in the PaletteModelController and the item index to its constructor. It will be needed because the item will be needed at the Palette level as well to show items which are not in a group. I had forgotten about that when writing earlier notes. 

Let's discuss if you have an alternative idea.